runtime/simulation/SimServer.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) 2014-2023 Embedded Systems and Applications, TU Darmstadt.
+# 
+# This file is part of TaPaSCo
+# (see https://github.com/esa-tu-darmstadt/tapasco).
+# 
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+# 
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import itertools
+from sys import byteorder
+from threading import Event
+import SimInterrupt
+from io import BytesIO
+from grpc_gen import sim_calls_pb2_grpc as sc_grpc
+from grpc_gen import sim_calls_pb2 as sc
+from grpc_gen import status_core_pb2 as status_core
+from SimInterrupt import SimInterrupt
+import cocotb
+import leb128
+from functools import reduce
+import queue
+from math import ceil
+import os
+
+def partition(l, size):
+    it = iter(l)
+    return iter(lambda: list(itertools.islice(it, size)), [])
+
+
+class SimServer(sc_grpc.SimRequestServicer):
+
```

Comment:
Please add a doc-string for this class and its public methods (including the constructor).

Reply:
The public methods are all implementations of the SimRequestServicer interface that is generated by grpc. Every public method coresponds to a service defined in runtime/libtapasco/protos/sim_calls.proto:8-17